### PR TITLE
Fill out validation.py and add tests

### PR DESCRIPTION
**Pull request by intrepiditee**
validation.py is for validating manifests and import specifications. Please see the required fields.

Thanks for reviewing.

### CODE REVIEW

File: import-automation/executor/app/executor/validation.py [hunk 1]
```diff
@@ -1,39 +1,193 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Functions for validating import targets specified in commit messages, manifests,
+and import specifications.
+"""
+
+import os
+import typing
```
Comment (jeffreyoldham) [inline import-automation/executor/app/executor/validation.py:21]: Sigh.  There is an exception: https://google.github.io/styleguide/pyguide.html#31912-imports-for-typing.

File: import-automation/executor/app/executor/validation.py [hunk 2]
```diff
@@ -1,39 +1,193 @@
...
+]
+_IMPORT_SPECIFICATION_REQUIRED_FIELDS = [
+    'import_name',
+    'provenance_url',
+    'provenance_description',
+    'curator_emails'
+]
+
+
+def task_info_valid(task_info: typing.Dict):
```
Comment (jeffreyoldham) [inline import-automation/executor/app/executor/validation.py:44]: Perhaps prepend with "is_" so it is clear this is a boolean function.

File: import-automation/executor/app/executor/validation.py [hunk 3]
```diff
@@ -1,39 +1,193 @@
...
+    Raises:
+        ValueError: The task info is not valid.
+    """
+    missing_keys = _get_missing_keys(_TASK_INFO_REQUIRED_FIELDS, task_info)
+    if missing_keys:
+        raise ValueError(f'Missing {utils.list_to_str(missing_keys)} in '
+                         f'task info {task_info}')
+
+
+def import_targets_valid(import_targets: typing.List[str],
```
Comment (jeffreyoldham) [inline import-automation/executor/app/executor/validation.py:61]: Likewise.

File: import-automation/executor/app/executor/validation.py [hunk 4]
```diff
@@ -1,39 +1,193 @@
...
+    for spec in manifest['import_specifications']:
+        _import_spec_valid(spec, repo_dir, import_dir)
+
+
+def _get_import_names_in_manifest(manifest):
+    return [spec.get('import_name')
+            for spec in manifest.get('import_specifications', [])]
+
+
+def _import_name_exists_in_manifest(
```
Comment (jeffreyoldham) [inline import-automation/executor/app/executor/validation.py:144]: Please add a function comment ala https://google.github.io/styleguide/pyguide.html#383-functions-and-methods.

File: import-automation/executor/app/executor/validation.py [hunk 5]
```diff
@@ -1,39 +1,193 @@
...
+    for absolute_name in absolute_names:
+        import_dir, import_name = import_target.split_absolute_import_name(
+            absolute_name)
+        _import_name_exists_in_manifest(
+            repo_dir, import_dir, import_name, manifest_filename)
 
-def manifest_valid(manifest):
-    pass
 
+def manifest_valid(
```
Comment (jeffreyoldham) [inline import-automation/executor/app/executor/validation.py:113]: Likewise.

File: import-automation/executor/test/test_validation.py [hunk 6]
```diff
@@ -0,0 +1,348 @@
...
+        with self.assertRaises(ValueError) as context:
+            validation._import_spec_valid(spec, self.repo_dir, 'scripts/us_fed')
+            self.assertIn('dir/foo.py, dir/../bar.py', str(context.exception))
+
+    def test_manifest_valid_fields_absent(self):
+        with self.assertRaises(ValueError) as context:
+            validation.manifest_valid({}, self.repo_dir, 'scripts/us_fed')
+            self.assertIn('import_specifications not found',
+                          str(context.exception))
+# Copyright 2020 Google LLC
```
Comment (jeffreyoldham) [inline import-automation/executor/test/test_validation.py:175]: Accidental duplication of the test code.

File: import-automation/executor/app/executor/validation.py (hunk 2 above)
Comment (intrepiditee) [reply import-automation/executor/app/executor/validation.py:44]: fixed

File: import-automation/executor/app/executor/validation.py (hunk 3 above)
Comment (intrepiditee) [reply import-automation/executor/app/executor/validation.py:61]: fixed

File: import-automation/executor/app/executor/validation.py (hunk 5 above)
Comment (intrepiditee) [reply import-automation/executor/app/executor/validation.py:113]: fixed

File: import-automation/executor/test/test_validation.py (hunk 6 above)
Comment (intrepiditee) [reply import-automation/executor/test/test_validation.py:175]: removed

File: import-automation/executor/app/executor/validation.py (hunk 4 above)
Comment (intrepiditee) [reply import-automation/executor/app/executor/validation.py:144]: added

**Comment by intrepiditee** [comment]
@jeffreyoldham Thanks. Updated.

File: import-automation/executor/app/executor/validation.py (hunk 1 above)
Comment (intrepiditee) [reply import-automation/executor/app/executor/validation.py:21]: I will continue to use typing.xxx

**Pull request merged**